Fix failing doctests in consumables module

## Description
Multiple doctests in `core/src/consumables/mod.rs` are failing. These tests use a `create_consumable` function that appears to be undefined in the test scope.

## Failing Doctests
1. `ConsumableSlots::add_consumable` (line 919)
2. `ConsumableSlots::clear_slot` (line 1077)
3. `ConsumableSlots::find_empty_slot` (line 1051)
4. `ConsumableSlots::get_consumable` (line 1001)
5. `ConsumableSlots::get_consumable_mut` (line 1028)
6. `ConsumableSlots::remove_consumable` (line 959)
7. `ConsumableSlots::iter` (line 1104)

## Error
```
error[E0425]: cannot find function `create_consumable` in this scope
```

## Likely Cause
The doctests are trying to use a helper function `create_consumable` that isn't available in the documentation test environment.

## Location
File: `core/src/consumables/mod.rs`

## Note
These failures are unrelated to PR #479 and appear to be pre-existing issues with the documentation tests.

--- a/core/src/consumables/mod.rs
+++ b/core/src/consumables/mod.rs
@@ -920,10 +920,10 @@ impl ConsumableSlots {
     /// use balatro_rs::consumables::{ConsumableSlots, SlotError};
     /// # use balatro_rs::consumables::{Consumable, ConsumableType, ConsumableEffect, TargetType, Target, ConsumableError};
     /// # use balatro_rs::game::Game;
-    /// # 
+    /// #
     /// # #[derive(Debug)]
     /// # struct MockConsumable;
-    /// # 
+    /// #
     /// # impl Consumable for MockConsumable {
     /// #     fn consumable_type(&self) -> ConsumableType { ConsumableType::Tarot }
     /// #     fn can_use(&self, _game_state: &Game, _target: &Target) -> bool { true }
@@ -932,7 +932,7 @@ impl ConsumableSlots {
     /// #     fn get_target_type(&self) -> TargetType { TargetType::None }
     /// #     fn get_effect_category(&self) -> ConsumableEffect { ConsumableEffect::Utility }
     /// # }
-    /// # 
+    /// #
     /// # fn create_consumable() -> Box<dyn Consumable> {
     /// #     Box::new(MockConsumable)
     /// # }
@@ -978,10 +978,10 @@ impl ConsumableSlots {
     /// use balatro_rs::consumables::{ConsumableSlots, SlotError};
     /// # use balatro_rs::consumables::{Consumable, ConsumableType, ConsumableEffect, TargetType, Target, ConsumableError};
     /// # use balatro_rs::game::Game;
-    /// # 
+    /// #
     /// # #[derive(Debug)]
     /// # struct MockConsumable;
-    /// # 
+    /// #
     /// # impl Consumable for MockConsumable {
     /// #     fn consumable_type(&self) -> ConsumableType { ConsumableType::Tarot }
     /// #     fn can_use(&self, _game_state: &Game, _target: &Target) -> bool { true }
@@ -990,7 +990,7 @@ impl ConsumableSlots {
     /// #     fn get_target_type(&self) -> TargetType { TargetType::None }
     /// #     fn get_effect_category(&self) -> ConsumableEffect { ConsumableEffect::Utility }
     /// # }
-    /// # 
+    /// #
     /// # fn create_consumable() -> Box<dyn Consumable> {
     /// #     Box::new(MockConsumable)
     /// # }
@@ -1038,10 +1038,10 @@ impl ConsumableSlots {
     /// use balatro_rs::consumables::ConsumableSlots;
     /// # use balatro_rs::consumables::{Consumable, ConsumableType, ConsumableEffect, TargetType, Target, ConsumableError};
     /// # use balatro_rs::game::Game;
-    /// # 
+    /// #
     /// # #[derive(Debug)]
     /// # struct MockConsumable;
-    /// # 
+    /// #
     /// # impl Consumable for MockConsumable {
     /// #     fn consumable_type(&self) -> ConsumableType { ConsumableType::Tarot }
     /// #     fn can_use(&self, _game_state: &Game, _target: &Target) -> bool { true }
@@ -1050,7 +1050,7 @@ impl ConsumableSlots {
     /// #     fn get_target_type(&self) -> TargetType { TargetType::None }
     /// #     fn get_effect_category(&self) -> ConsumableEffect { ConsumableEffect::Utility }
     /// # }
-    /// # 
+    /// #
     /// # fn create_consumable() -> Box<dyn Consumable> {
     /// #     Box::new(MockConsumable)
     /// # }
@@ -1083,10 +1083,10 @@ impl ConsumableSlots {
     /// use balatro_rs::consumables::ConsumableSlots;
     /// # use balatro_rs::consumables::{Consumable, ConsumableType, ConsumableEffect, TargetType, Target, ConsumableError};
     /// # use balatro_rs::game::Game;
-    /// # 
+    /// #
     /// # #[derive(Debug)]
     /// # struct MockConsumable;
-    /// # 
+    /// #
     /// # impl Consumable for MockConsumable {
     /// #     fn consumable_type(&self) -> ConsumableType { ConsumableType::Tarot }
     /// #     fn can_use(&self, _game_state: &Game, _target: &Target) -> bool { true }
@@ -1095,7 +1095,7 @@ impl ConsumableSlots {
     /// #     fn get_target_type(&self) -> TargetType { TargetType::None }
     /// #     fn get_effect_category(&self) -> ConsumableEffect { ConsumableEffect::Utility }
     /// # }
-    /// # 
+    /// #
     /// # fn create_consumable() -> Box<dyn Consumable> {
     /// #     Box::new(MockConsumable)
     /// # }
@@ -1124,10 +1124,10 @@ impl ConsumableSlots {
     /// use balatro_rs::consumables::ConsumableSlots;
     /// # use balatro_rs::consumables::{Consumable, ConsumableType, ConsumableEffect, TargetType, Target, ConsumableError};
     /// # use balatro_rs::game::Game;
-    /// # 
+    /// #
     /// # #[derive(Debug)]
     /// # struct MockConsumable;
-    /// # 
+    /// #
     /// # impl Consumable for MockConsumable {
     /// #     fn consumable_type(&self) -> ConsumableType { ConsumableType::Tarot }
     /// #     fn can_use(&self, _game_state: &Game, _target: &Target) -> bool { true }
@@ -1136,7 +1136,7 @@ impl ConsumableSlots {
     /// #     fn get_target_type(&self) -> TargetType { TargetType::None }
     /// #     fn get_effect_category(&self) -> ConsumableEffect { ConsumableEffect::Utility }
     /// # }
-    /// # 
+    /// #
     /// # fn create_consumable() -> Box<dyn Consumable> {
     /// #     Box::new(MockConsumable)
     /// # }
@@ -1168,10 +1168,10 @@ impl ConsumableSlots {
     /// use balatro_rs::consumables::{ConsumableSlots, SlotError};
     /// # use balatro_rs::consumables::{Consumable, ConsumableType, ConsumableEffect, TargetType, Target, ConsumableError};
     /// # use balatro_rs::game::Game;
-    /// # 
+    /// #
     /// # #[derive(Debug)]
     /// # struct MockConsumable;
-    /// # 
+    /// #
     /// # impl Consumable for MockConsumable {
     /// #     fn consumable_type(&self) -> ConsumableType { ConsumableType::Tarot }
     /// #     fn can_use(&self, _game_state: &Game, _target: &Target) -> bool { true }
@@ -1180,7 +1180,7 @@ impl ConsumableSlots {
     /// #     fn get_target_type(&self) -> TargetType { TargetType::None }
     /// #     fn get_effect_category(&self) -> ConsumableEffect { ConsumableEffect::Utility }
     /// # }
-    /// # 
+    /// #
     /// # fn create_consumable() -> Box<dyn Consumable> {
     /// #     Box::new(MockConsumable)
     /// # }
@@ -1213,10 +1213,10 @@ impl ConsumableSlots {
     /// use balatro_rs::consumables::ConsumableSlots;
     /// # use balatro_rs::consumables::{Consumable, ConsumableType, ConsumableEffect, TargetType, Target, ConsumableError};
     /// # use balatro_rs::game::Game;
-    /// # 
+    /// #
     /// # #[derive(Debug)]
     /// # struct MockConsumable;
-    /// # 
+    /// #
     /// # impl Consumable for MockConsumable {
     /// #     fn consumable_type(&self) -> ConsumableType { ConsumableType::Tarot }
     /// #     fn can_use(&self, _game_state: &Game, _target: &Target) -> bool { true }
@@ -1225,7 +1225,7 @@ impl ConsumableSlots {
     /// #     fn get_target_type(&self) -> TargetType { TargetType::None }
     /// #     fn get_effect_category(&self) -> ConsumableEffect { ConsumableEffect::Utility }
     /// # }
-    /// # 
+    /// #
     /// # fn create_consumable() -> Box<dyn Consumable> {
     /// #     Box::new(MockConsumable)
     /// # }

